runtime/lib/capi/RuntimeCAPI.cpp

```diff
@@ -691,20 +691,14 @@ void __quantum__qis__Probs(MemRefT_double_1d *result, int64_t numQubits, ...)
         numQubits = __quantum__rt__num_qubits();
     }
 
-    std::vector<double> sv_probs;
-
     if (wires.empty()) {
-        sv_probs = Catalyst::Runtime::CAPI::DRIVER->get_device()->Probs();
+        Catalyst::Runtime::CAPI::DRIVER->get_device()->Probs(
+            std::span{result_p->data_allocated, result_p->sizes[0]});
```

Comment:
How are the strides taken into account?

Reply:
Not at the moment and didn't run into any issues/bugs. let's chat about it :)

Reply:
What was the result of this discussion?

Reply:
We're going to use `MemRefView`s instead of `std::span` to tackle the issue of not considering strides for 1-dim and 2-dim memrefs.

Reply:
Could this be useful? In the `mlir_c_runner` library there are iterators for memrefs that take into account strides.

```cpp
/// Iterate over all elements in a dynamic memref.
template <typename T>
class DynamicMemRefIterator {
public:
  using iterator_category = std::forward_iterator_tag;
  using value_type = T;
  using difference_type = std::ptrdiff_t;
  using pointer = T *;
  using reference = T &;

  DynamicMemRefIterator(DynamicMemRefType<T> &descriptor, int64_t offset = 0)
      : offset(offset), descriptor(&descriptor) {
    indices.resize(descriptor.rank, 0);
  }

  DynamicMemRefIterator<T> &operator++() {
    if (descriptor->rank == 0) {
      offset = -1;
      return *this;
    }

    int dim = descriptor->rank - 1;

    while (dim >= 0 && indices[dim] == (descriptor->sizes[dim] - 1)) {
      offset -= indices[dim] * descriptor->strides[dim];
      indices[dim] = 0;
      --dim;
    }

    if (dim < 0) {
      offset = -1;
      return *this;
    }
    
    // ... snip ...
```

Reply:
Oh cool, I was curious about this as well

Reply:
Oh! I didn't know this iterator exists. Thank you for pointing it out. I'll take a look and it can definitely simplify the code.

Reply:
@maliasadi note that we probably don't want to add the MLIR project as a dependency to the runtime module  though

Reply:
@erick-xanadu, just got back to this PR from other tasks and preparing for the presentation. 
Thanks again for sharing! I looked at `mlir_c_runner`. Although our custom iterator has similarities with `DynamicMemRefIterator` technically, we shouldn't necessarily need any code from `mlir_c_runner` using `MemRefView`. Let me create a new PR with the proposed solution and I think it should be fine.

@dime10 I wouldn't let that happen. Not any chance of adding this dependency to the runtime (for now) :sweat_smile: